Shoko.TrayService/App.xaml.cs

```diff
@@ -1,174 +1,167 @@
+        => e.Cancel = true;
+    
+    private void OnInstanceOnServerShutdown(object? o, EventArgs eventArgs) 
+        => Shutdown();
 
-            if (!ServerSettings.Instance.FirstRun)
-                ShokoServer.RunWorkSetupDB();
-            else Logger.Warn("The Server is NOT STARTED. It needs to be configured via webui or the settings.json");
+    private static void OnInstanceOnPropertyChanged(object? _, PropertyChangedEventArgs e)
+    {
+        if (e.PropertyName == "StartupFailedMessage" && ServerState.Instance.StartupFailed)
+            Console.WriteLine("Startup failed! Error message: " + ServerState.Instance.StartupFailedMessage);
+    }
 
-            ShokoServer.Instance.ServerShutdown += (sender, eventArgs) => Shutdown();
-            Utils.YesNoRequired += (sender, e) =>
-            {
-                e.Cancel = true;
-            };
+    private void InitialiseTaskbarIcon()
+    {
+        _icon = new TaskbarIcon{
+                                   ToolTipText = "Shoko Server", ContextMenu = CreateContextMenu(), MenuActivation = PopupActivationMode.All, Visibility = Visibility.Visible,
+                               };
+        using var iconStream = GetResourceStream(new Uri("pack://application:,,,/ShokoServer;component/db.ico"))?.Stream;
+        if (iconStream is not null)
+            _icon.Icon = new Icon(iconStream);
+    }
+    
+    private ContextMenu CreateContextMenu()
+    {
+        var menu  = new ContextMenu();
+        var webui = new MenuItem{Header = "Open WebUI"};
+        webui.Click += OnWebuiOpenWebUIClick;
+        menu.Items.Add(webui);
+        webui       =  new MenuItem{Header = "Exit"};
+        webui.Click += OnWebuiExit;
+        menu.Items.Add(webui);
+        return menu;
+    }
+    
+    private void OnWebuiExit(object? sender, RoutedEventArgs args) 
+        => Dispatcher.Invoke(Shutdown);
 
-            ServerState.Instance.PropertyChanged += (sender, e) =>
-            {
-                if (e.PropertyName == "StartupFailedMessage" && ServerState.Instance.StartupFailed)
-                {
-                    Console.WriteLine("Startup failed! Error message: " + ServerState.Instance.StartupFailedMessage);
-                }
-            };
-            ShokoService.CmdProcessorGeneral.OnQueueStateChangedEvent +=
-                ev => Console.WriteLine($"General Queue state change: {ev.QueueState.formatMessage()}");
+    private static void OnWebuiOpenWebUIClick(object? sender, RoutedEventArgs args)
+    {
+        string? ip = null;
+        try
+        {
+            ip = GetLocalIPv6();
         }
+        catch (Exception) { /*ok no IPv6 available.*/ }
 
-        protected override void OnExit(ExitEventArgs e)
+        if (string.IsNullOrWhiteSpace(ip))
+            ip = GetLocalIpv4OrFallback();
+        else
+            ip = $"[{ip}]";
+        
+        var url = $"http://{ip}:{ServerSettings.Instance.ServerPort}";
```

Comment:
Instead of doing this whole thing... why not just `http://localhost:{ServerSettings.Instance.ServerPort}`